### DISCUSSION TITLE
Proposal: Enable Toolbar Buttons to use dynamic disabled status

### DISCUSSION THREAD

**Discussion by pmario** [TiddlyWiki/TiddlyWiki5#5281] [Ideas]
In TW 5.1.23 the button widget gets a "disabled" attribute. 

This information can be used by CSS with `button:disabled{fill:#aaa;}`. ... ATM toolbar buttons don't have a possibility to find out, if they should set the disabled attribute. Toolbars heavily depend on "config tiddlers". So everything needs to be defined dynamically.

The button config tiddlers have a field `condition`, which allows them to know if they should be visible. ... 

## The proposal is, to use a similar mechanism to define "disabled"

Add a new field `condition-disabled` to the button-config tiddler. It should contain a filter, that allows the button to set the `disabled` attribute. 

The implementation is relatively simple and I can create a PR really fast, since I have a working implementation, that I want to use for one of my plugins. 

@Jermolene, @saqimtiaz, @BurningTreeC and others ... what do you think?

`$:/core/ui/EditTemplate/body/toolbar/button` only need 2 `<$set name=disabled filter={{!!condition-disabled}}>` and we need some CSS adjustments as mentioned above. 

Most work will be needed for the docs.

**Comment by Jermolene** [comment]
Hi @pmario I agree, good idea. We'd also need to add palette entries for styling disabled buttons and the SVGs within them.

**Reply by pmario** [reply]
I'll create a PR. There is only a little bit of styling needed.

**Reply by pmario** [reply]
We already have a `<<muted-foreground>>` setting for SVG icons within buttons

**Comment by pmario** [comment]
@Jermolene ... There is no docs, how to create a new toolbar button. ... Should it be part of TW-docs or TW-DEV docs. ... It's a relatively complex configuration.

**Reply by Jermolene** [reply]
Hi @pmario I think the distinction between the "dev" community and the mainstream community is basically JavaScript. Similarly, https://tiddlywiki.com/dev is extra information for JavaScript developers that we don't want getting in the way on the main site.

So, I think the docs for cresting a new toolbar button belong in the main docs.
